packages/blade/src/components/Typography/Heading/Heading.tsx

```diff
@@ -0,0 +1,85 @@
+import type { ReactElement } from 'react';
+import type { TextTypes } from '../../../tokens/theme/theme';
+import getPlatformType from '../../../utils/getPlatformType';
+import type { Theme } from '../../BladeProvider';
+import BaseText from '../BaseText';
+import type { BaseTextProps } from '../BaseText/BaseText';
+
+type HeadingCommonProps = {
+  type: TextTypes;
+  children?: React.ReactNode;
+};
+
+type HeadingNormalVariant = HeadingCommonProps & {
+  variant: 'small' | 'medium' | 'large';
+  weight: keyof Theme['typography']['fonts']['weight'];
+};
+
+type HeadingSubHeadingVariant = HeadingCommonProps & {
+  variant: 'subheading';
+  weight: keyof Pick<Theme['typography']['fonts']['weight'], 'bold'>;
+};
+
+/**
+ * Conditionally changing props based on variant.
+ * Overloads or union gives wrong intellisense.
+ */
+export type HeadingProps<T> = T extends {
+  variant: infer Variant;
+}
+  ? Variant extends 'small' | 'medium' | 'large'
+    ? HeadingNormalVariant
+    : Variant extends 'subheading'
+    ? HeadingSubHeadingVariant
+    : T
+  : T;
+
+const getProps = <T extends { variant: 'small' | 'medium' | 'large' | 'subheading' }>({
+  variant,
+  type,
+  weight,
+}: Pick<HeadingProps<T>, 'variant' | 'type' | 'weight'>): BaseTextProps => {
+  const props: BaseTextProps = {
+    color: `surface.text.${type}.lowContrast`,
+    fontSize: 25,
+    fontWeight: weight,
+    fontStyle: 'normal',
+    lineHeight: 'm',
+    fontFamily: 'text',
+  };
+
+  if (variant === 'small') {
+    props.fontSize = 200;
+    props.lineHeight = 'xl';
+    props.as = getPlatformType() === 'browser' || getPlatformType() === 'node' ? 'h6' : undefined;
```

Comment:
Extract this to a local var, we repeating it a lot. 

```ts
const isBrowserLike = getPlatformType() === 'browser' || getPlatformType() === 'node'; 
```

Reply:
Or make this into another util/helper inside `getPlatformType`? Seen this being used in [Title.tsx](https://github.com/razorpay/blade/blob/master/packages/blade/src/components/Typography/Title/Title.tsx#L26) and imagine it will be used extensively in other components too.